### PR TITLE
Raise a ``UserWarning`` when ``max_calls_total < 2``

### PR DESCRIPTION
**Pull request by DanielNoord**
And also fix the date in the Changelog of ``3.0.0``.

### CODE REVIEW

**Comment by jochemb** [comment]
There is a good use case for `max_calls_total = 1` though. People use dynamic globals to prevent retry behavior in their tests. This was explicitly added as a feature in https://github.com/channable/opnieuw/pull/3

So I don't think we want to limit usage in this way :/

**Comment by DanielNoord** [comment]
> There is a good use case for `max_calls_total = 1` though. People use dynamic globals to prevent retry behavior in their tests. This was explicitly added as a feature in #3
> 
> So I don't think we want to limit usage in this way :/

They can also disable erroring out on this `UserWarning` then right? Since it's a warning it won't crash the code.

**Comment by DanielNoord** [comment]
@jochemb A `UserWarning` does show up during normal execution but does not count as an `Exception`. It's more like a logging statement.

We could consider using just `Warning`, but I don't really know what's best. It feels like `UserWarning` fits best with our intention here.

File: opnieuw/retries.py [hunk 1]
```diff
@@ -209,6 +210,16 @@ def foo() -> None:
         https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/
     """
 
+    if max_calls_total < 2:
+        warnings.warn(
+            "max_calls_total should at least be 2 to make opnieuw retry once. "
+            f"It is set to '{max_calls_total}'. If you want to retry immediately "
+            "consider using `opnieuw.test_util.retry_immediately`, if you don't want "
+            "any retries considering using `opnieuw.util.no_retries`.",
```
[the remark this answers is not in the document]
Comment (DanielNoord) [reply opnieuw/retries.py]: Yeah, that's better! Fixed!

**Review by jochemb — APPROVED** [review]
Thanks for the addition :safety_vest:

**Comment by DanielNoord** [comment]
@OpsBotPrime merge

**Comment by OpsBotPrime** [comment]
Pull request approved for merge by @DanielNoord, rebasing now.

**Comment by OpsBotPrime** [comment]
Rebased as f558ae6994d0371b7c54db29b554893e243eb2c5, waiting for CI …

**Comment by OpsBotPrime** [comment]
[CI job :yellow_circle:](https://channable.semaphoreci.com/workflows/7a481c90-2ce0-4add-9668-d665f10ac55b?pipeline_id=1e08898e-df5f-4823-9086-e78f40df5f50) started.

**Pull request merged**